<a href="https://colab.research.google.com/github/CoWoGeo/PUS2022_CWolk/blob/main/ProjectLathamWolk/LathamWolkMidtermData.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
!pip install geopandas

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 1.0 MB 4.2 MB/s 
     |████████████████████████████████| 16.7 MB 59.7 MB/s 
     |████████████████████████████████| 6.3 MB 41.6 MB/s 


In [2]:
import geopandas as gpd

In [35]:
import numpy as np

In [5]:
import pandas as pd

In [4]:
licenses = gpd.GeoDataFrame.from_file("https://phl.carto.com/api/v2/sql?q=SELECT+*+FROM+business_licenses&filename=business_licenses&format=shp&skipfields=cartodb_id")

In [9]:
licenses.head()

objectid  addressobj             address unit_type unit_num         zip  \
0        27  15326419.0      1816 MARKET ST         #       G1  19103-3717   
1        28  15412910.0  3800 LANCASTER AVE         #       2A  19104-2318   
2        29  15878447.0     7673 WINSTON RD      None     None  19118-3528   
3        30  15753726.0   4200 WOODLAND AVE      None     None  19104-4446   
4        31  15494453.0     4600 N BROAD ST      None     None  19140-1218   

  censustrac parcel_id_ opa_accoun                           opa_owner  ...  \
0       4.02      19002       None                                None  ...   
1         91      90715  871286150                           CHEN RONG  ...   
2        257      56632  882970820      MEKIS JOSEPH A, MEKIS VERONICA  ...   
3         77     589969  773501000  PHILA COLLEGE OF, PHARMACY SCIENCE  ...   
4        281      10441  882929625                       SIDNEY BOOKER  ...   

  ownercon_5 ownercon_6 ownercon_7 ownercon_8 ownercon_9     geocode_x  \
0       None       None       None       None       None  2.691389e+06   
1       None       None       None       None       None  2.684032e+06   
2       None       None       None       None       None  2.683017e+06   
3       None       None       None       None       None  2.681397e+06   
4       None       None       None       None       None  2.697010e+06   

       geocode_y council_di posse_jobi                    geometry  
0  236268.605606          5    9181088  POINT (-75.17115 39.95294)  
1  238666.281821          3    9181194  POINT (-75.19713 39.96010)  
2  277924.597766          8    9181195  POINT (-75.19671 40.06791)  
3  233572.767812          3    9181196  POINT (-75.20705 39.94633)  
4  262120.819110          8    9181197  POINT (-75.14840 40.02342)  

[5 rows x 42 columns]

In [10]:
licenses.columns()

TypeError: ignored

In [11]:
licenses

objectid   addressobj             address unit_type unit_num  \
0             27   15326419.0      1816 MARKET ST         #       G1   
1             28   15412910.0  3800 LANCASTER AVE         #       2A   
2             29   15878447.0     7673 WINSTON RD      None     None   
3             30   15753726.0   4200 WOODLAND AVE      None     None   
4             31   15494453.0     4600 N BROAD ST      None     None   
...          ...          ...                 ...       ...      ...   
393346    391987  133999146.0      2430 N 25TH ST      None     None   
393347    391988  135245705.0      4801 N 11TH ST         #        2   
393348    391989  138831647.0       627 N 18TH ST         #      10B   
393349    391990  130053868.0  1244 N LAWRENCE ST      None     None   
393350    391991   15529368.0  5334 HAVERFORD AVE      None     None   

               zip censustrac parcel_id_ opa_accoun  \
0       19103-3717       4.02      19002       None   
1       19104-2318         91      90715  871286150   
2       19118-3528        257      56632  882970820   
3       19104-4446         77     589969  773501000   
4       19140-1218        281      10441  882929625   
...            ...        ...        ...        ...   
393346  19132-4228     169.01     149011  281033805   
393347  19141-3403        283     518077  491398000   
393348  19130-3344     134.02     481774       None   
393349  19122-4428        144     180519  182250400   
393350  19139-1407         93     330626  441048500   

                                  opa_owner  ... ownercon_5 ownercon_6  \
0                                      None  ...       None       None   
1                                 CHEN RONG  ...       None       None   
2            MEKIS JOSEPH A, MEKIS VERONICA  ...       None       None   
3        PHILA COLLEGE OF, PHARMACY SCIENCE  ...       None       None   
4                             SIDNEY BOOKER  ...       None       None   
...                                     ...  ...        ...        ...   
393346                      VMR CORPORATION  ...       None       None   
393347  KS INVESTMENTS AND ACQUISITIONS LLC  ...       None       None   
393348                                 None  ...       None       None   
393349             RODRIGUEZ JESSICA AMANDA  ...       None       None   
393350      BUSSEY SHANNON, SIMMONS HANESHA  ...       None       None   

       ownercon_7 ownercon_8 ownercon_9     geocode_x      geocode_y  \
0            None       None       None  2.691389e+06  236268.605606   
1            None       None       None  2.684032e+06  238666.281821   
2            None       None       None  2.683017e+06  277924.597766   
3            None       None       None  2.681397e+06  233572.767812   
4            None       None       None  2.697010e+06  262120.819110   
...           ...        ...        ...           ...            ...   
393346       None       None       None  2.690335e+06  250664.787664   
393347       None       None       None  2.698790e+06  262941.153801   
393348       None       None       None  2.692465e+06  241319.777585   
393349       None       None       None  2.698799e+06  243155.050598   
393350       None       None       None  2.675512e+06  240450.228789   

       council_di posse_jobi                    geometry  
0               5    9181088  POINT (-75.17115 39.95294)  
1               3    9181194  POINT (-75.19713 39.96010)  
2               8    9181195  POINT (-75.19671 40.06791)  
3               3    9181196  POINT (-75.20705 39.94633)  
4               8    9181197  POINT (-75.14840 40.02342)  
...           ...        ...                         ...  
393346          5  518575422  POINT (-75.17341 39.99252)  
393347          8  518586439  POINT (-75.14195 40.02553)  
393348          5  518599274  POINT (-75.16679 39.96671)  
393349          5  518604366  POINT (-75.14401 39.97124)  
393350          3  518609845  POINT (-75.22733 39.96567)  

[393351 rows x 42 c

In [12]:
print(licenses.columns.tolist())

['objectid', 'addressobj', 'address', 'unit_type', 'unit_num', 'zip', 'censustrac', 'parcel_id_', 'opa_accoun', 'opa_owner', 'licensenum', 'revenuecod', 'licensetyp', 'rentalcate', 'initialiss', 'mostrecent', 'expiration', 'inactiveda', 'licensesta', 'numberofun', 'owneroccup', 'legalfirst', 'legallastn', 'legalname', 'legalentit', 'business_n', 'business_m', 'ownerconta', 'ownercon_1', 'ownercon_2', 'ownercon_3', 'ownercon_4', 'ownercon_5', 'ownercon_6', 'ownercon_7', 'ownercon_8', 'ownercon_9', 'geocode_x', 'geocode_y', 'council_di', 'posse_jobi', 'geometry']


In [13]:
licenses.info

<class 'geopandas.geodataframe.GeoDataFrame'>
RangeIndex: 393351 entries, 0 to 393350
Data columns (total 42 columns):
 #   Column      Non-Null Count   Dtype   
---  ------      --------------   -----   
 0   objectid    393351 non-null  int64   
 1   addressobj  373633 non-null  float64 
 2   address     376165 non-null  object  
 3   unit_type   115488 non-null  object  
 4   unit_num    114961 non-null  object  
 5   zip         376139 non-null  object  
 6   censustrac  375641 non-null  object  
 7   parcel_id_  374332 non-null  object  
 8   opa_accoun  366104 non-null  object  
 9   opa_owner   365848 non-null  object  
 10  licensenum  393351 non-null  object  
 11  revenuecod  393351 non-null  object  
 12  licensetyp  393351 non-null  object  
 13  rentalcate  228235 non-null  object  
 14  initialiss  393351 non-null  object  
 15  mostrecent  393351 non-null  object  
 16  expiration  384272 non-null  object  
 17  inactiveda  226649 non-null  object  
 18  licensesta  3933

# <class 'geopandas.geodataframe.GeoDataFrame'>
RangeIndex: 393351 entries, 0 to 393350
Data columns (total 42 columns):
 #   Column      Non-Null Count   Dtype   
---  ------      --------------   -----   
 0   objectid    393351 non-null  int64   
 1   addressobj  373633 non-null  float64 
 2   address     376165 non-null  object  
 3   unit_type   115488 non-null  object  
 4   unit_num    114961 non-null  object  
 5   zip         376139 non-null  object  
 6   censustrac  375641 non-null  object  
 7   parcel_id_  374332 non-null  object  
 8   opa_accoun  366104 non-null  object  
 9   opa_owner   365848 non-null  object  
 10  licensenum  393351 non-null  object  
 11  revenuecod  393351 non-null  object  
 12  licensetyp  393351 non-null  object  
 13  rentalcate  228235 non-null  object  
 14  initialiss  393351 non-null  object  
 15  mostrecent  393351 non-null  object  
 16  expiration  384272 non-null  object  
 17  inactiveda  226649 non-null  object  
 18  licensesta  393351 non-null  object  
 19  numberofun  271504 non-null  float64 
 20  owneroccup  249032 non-null  object  
 21  legalfirst  219854 non-null  object  
 22  legallastn  219849 non-null  object  
 23  legalname   392837 non-null  object  
 24  legalentit  392836 non-null  object  
 25  business_n  392913 non-null  object  
 26  business_m  391856 non-null  object  
 27  ownerconta  9601 non-null    object  
 28  ownercon_1  9595 non-null    object  
 29  ownercon_2  9601 non-null    object  
 30  ownercon_3  9556 non-null    object  
 31  ownercon_4  9601 non-null    object  
 32  ownercon_5  893 non-null     object  
 33  ownercon_6  893 non-null     object  
 34  ownercon_7  911 non-null     object  
 35  ownercon_8  950 non-null     object  
 36  ownercon_9  911 non-null     object  
 37  geocode_x   375639 non-null  float64 
 38  geocode_y   375639 non-null  float64 
 39  council_di  375633 non-null  object  
 40  posse_jobi  393351 non-null  object  
 41  geometry    375639 non-null  geometry
dtypes: float64(4), geometry(1), int64(1), object(36)
memory usage: 126.0+ MB

In [ ]:
# licenses.info()

<class 'geopandas.geodataframe.GeoDataFrame'>
RangeIndex: 393351 entries, 0 to 393350
Data columns (total 42 columns):
 #   Column      Non-Null Count   Dtype   
---  ------      --------------   -----   
 0   objectid    393351 non-null  int64   
 1   addressobj  373633 non-null  float64 
 2   address     376165 non-null  object  
 3   unit_type   115488 non-null  object  
 4   unit_num    114961 non-null  object  
 5   zip         376139 non-null  object  
 6   censustrac  375641 non-null  object  
 7   parcel_id_  374332 non-null  object  
 8   opa_accoun  366104 non-null  object  
 9   opa_owner   365848 non-null  object  
 10  licensenum  393351 non-null  object  
 11  revenuecod  393351 non-null  object  
 12  licensetyp  393351 non-null  object  
 13  rentalcate  228235 non-null  object  
 14  initialiss  393351 non-null  object  
 15  mostrecent  393351 non-null  object  
 16  expiration  384272 non-null  object  
 17  inactiveda  226649 non-null  object  
 18  licensesta  3933

In [16]:
firstviolations = gpd.GeoDataFrame.from_file("https://phl.carto.com/api/v2/sql?filename=violations&format=shp&skipfields=cartodb_id&q=SELECT%20*%20FROM%20violations%20WHERE%20violationdate%20%3E=%20%272013-01-01%27%20AND%20violationdate%20%3C%20%272016-01-01%27")

In [17]:
firstviolations.info()

<class 'geopandas.geodataframe.GeoDataFrame'>
RangeIndex: 323545 entries, 0 to 323544
Data columns (total 31 columns):
 #   Column      Non-Null Count   Dtype   
---  ------      --------------   -----   
 0   objectid    323545 non-null  int64   
 1   addressobj  323545 non-null  object  
 2   parcel_id_  26 non-null      object  
 3   casenumber  323545 non-null  object  
 4   casecreate  311564 non-null  object  
 5   casecomple  268734 non-null  object  
 6   casetype    26 non-null      object  
 7   casestatus  323545 non-null  object  
 8   caserespon  311554 non-null  object  
 9   casepriori  311544 non-null  object  
 10  violationn  323545 non-null  object  
 11  violationd  323545 non-null  object  
 12  violationc  323545 non-null  object  
 13  violatio_1  323452 non-null  object  
 14  violations  318967 non-null  object  
 15  violationr  5 non-null       object  
 16  violatio_2  5 non-null       object  
 17  mostrecent  323354 non-null  object  
 18  opa_accoun  3200

In [19]:
firstviolations

objectid addressobj parcel_id_ casenumber  casecreate  casecomple  \
0            960  134326208      40813     574371  2017-03-01  2021-03-31   
1           1372   15873319      64089     575282  2017-03-10        None   
2           1373   15873319      64089     575282  2017-03-10        None   
3           1374   15873319      64089     575282  2017-03-10        None   
4           1375   15873319      64089     575282  2017-03-10        None   
...          ...        ...        ...        ...         ...         ...   
323540   1639658      99953       None     401313  2013-09-10  2014-01-15   
323541   1639662      99955       None     400064  2013-08-29  2013-09-13   
323542   1639663      99955       None     400064  2013-08-29  2013-09-13   
323543   1639664      99955       None     400064  2013-08-29  2013-09-13   
323544   1638701      99789       None     467778  2015-02-04  2015-02-19   

                   casetype    casestatus                     caserespon  \
0       NOTICE OF VIOLATION        CLOSED  CODE ENFORCEMENT INVESTIGATOR   
1       NOTICE OF VIOLATION  IN VIOLATION          BUILDING INVESTIGATOR   
2       NOTICE OF VIOLATION  IN VIOLATION          BUILDING INVESTIGATOR   
3       NOTICE OF VIOLATION  IN VIOLATION          BUILDING INVESTIGATOR   
4       NOTICE OF VIOLATION  IN VIOLATION          BUILDING INVESTIGATOR   
...                     ...           ...                            ...   
323540                 None        CLOSED                           CLIP   
323541                 None        CLOSED                           CLIP   
323542                 None        CLOSED                           CLIP   
323543                 None        CLOSED                           CLIP   
323544                 None        CLOSED                           CLIP   

       casepriori  ... unit_num         zip censustrac  \
0        STANDARD  ...     None       19111        306   
1        STANDARD  ...     None  19121-1815     151.02   
2        STANDARD  ...     None  19121-1815     151.02   
3        STANDARD  ...     None  19121-1815     151.02   
4        STANDARD  ...     None  19121-1815     151.02   
...           ...  ...      ...         ...        ...   
323540   STANDARD  ...     None  19111-3647        338   
323541   STANDARD  ...     None  19111-3647        338   
323542   STANDARD  ...     None  19111-3647        338   
323543   STANDARD  ...     None  19111-3647        338   
323544   STANDARD  ...     None  19136-2123        349   

                                  opa_owner systemofre     geocode_x  \
0                     YANOVSKY FAMILY TRUST    ECLIPSE  2.712031e+06   
1                      WILLIAMS JOSEPH T II    ECLIPSE  2.687599e+06   
2                      WILLIAMS JOSEPH T II    ECLIPSE  2.687599e+06   
3                      WILLIAMS JOSEPH T II    ECLIPSE  2.687599e+06   
4                      WILLIAMS JOSEPH T II    ECLIPSE  2.687599e+06   
...                                     ...        ...           ...   
323540            LIBERTY REALTY MANAGEMENT     HANSEN  2.715473e+06   
323541                     FARBER BETTE LEE     HANSEN  2.715524e+06   
323542                     FARBER BETTE LEE     HANSEN  2.715524e+06   
323543                     FARBER BETTE LEE     HANSEN  2.715524e+06   
323544  RABOY RANDY L            PATRICIA M     HANSEN  2.734709e+06   

            geocode_y council_di posse_jobi                    geometry  
0       271297.694077          9  195195074  POINT (-75.09380 40.04737)  
1       247416.020590          5  195195263  POINT (-75.18351 39.98383)  
2       247416.020590          5  195195263  POINT (-75.18351 39.98383)  
3       247416.020590          5  195195263  POINT (-75.18351 39.98383)  
4       247416.020590          5  195195263  POINT (-75.18351 39.98383)  
...               ...        ...        ...                         ...  
323540  276062.435571       None       None  POINT (-75.08099 40.06016)  
323541  276027.6

In [21]:
secondviolations = gpd.GeoDataFrame.from_file("https://phl.carto.com/api/v2/sql?filename=violations&format=shp&skipfields=cartodb_id&q=SELECT%20*%20FROM%20violations%20WHERE%20violationdate%20%3E=%20%272016-01-01%27%20AND%20violationdate%20%3C%20%272019-01-01%27")

In [22]:
recentviolations = gpd.GeoDataFrame.from_file("https://phl.carto.com/api/v2/sql?filename=violations&format=shp&skipfields=cartodb_id&q=SELECT * FROM violations WHERE violationdate >= '2019-01-01'")

InvalidURL: ignored

In [23]:
secondviolations

objectid addressobj parcel_id_ casenumber  casecreate  casecomple  \
0             51  163531503     532397     569117  2017-01-10        None   
1             52  163531503     532397     569117  2017-01-10        None   
2             53  163531503     532397     569117  2017-01-10        None   
3             54  163531503     532397     569117  2017-01-10        None   
4             55  163531503     532397     569117  2017-01-10        None   
...          ...        ...        ...        ...         ...         ...   
333779   1639666      99955       None     584921  2017-05-22  2017-06-12   
333780   1639670      99956       None     540446  2016-06-20  2016-08-04   
333781   1639671      99956       None     540446  2016-06-20  2016-08-04   
333782   1639816      99998       None     528521  2016-04-19        None   
333783   1639817      99998       None     583448  2017-05-11        None   

                   casetype    casestatus                     caserespon  \
0       NOTICE OF VIOLATION  IN VIOLATION  CODE ENFORCEMENT INVESTIGATOR   
1       NOTICE OF VIOLATION  IN VIOLATION  CODE ENFORCEMENT INVESTIGATOR   
2       NOTICE OF VIOLATION  IN VIOLATION  CODE ENFORCEMENT INVESTIGATOR   
3       NOTICE OF VIOLATION  IN VIOLATION  CODE ENFORCEMENT INVESTIGATOR   
4       NOTICE OF VIOLATION  IN VIOLATION  CODE ENFORCEMENT INVESTIGATOR   
...                     ...           ...                            ...   
333779                 None        CLOSED                           CLIP   
333780                 None        CLOSED                           CLIP   
333781                 None        CLOSED                           CLIP   
333782                 None        CLOSED                             DE   
333783                 None        CLOSED                             CI   

                   casepriori  ... unit_num         zip censustrac  \
0                    STANDARD  ...     None  19144-2614        238   
1                    STANDARD  ...     None  19144-2614        238   
2                    STANDARD  ...     None  19144-2614        238   
3                    STANDARD  ...     None  19144-2614        238   
4                    STANDARD  ...     None  19144-2614        238   
...                       ...  ...      ...         ...        ...   
333779               STANDARD  ...     None  19111-3647        338   
333780               STANDARD  ...     None  19111-3647        338   
333781               STANDARD  ...     None  19111-3647        338   
333782  CONSTRUCTION SERVICES  ...     None  19111-3605        338   
333783               STANDARD  ...     None  19111-3605        338   

                  opa_owner systemofre     geocode_x      geocode_y  \
0              SAMUEL PETER    ECLIPSE  2.687111e+06  268093.728721   
1              SAMUEL PETER    ECLIPSE  2.687111e+06  268093.728721   
2              SAMUEL PETER    ECLIPSE  2.687111e+06  268093.728721   
3              SAMUEL PETER    ECLIPSE  2.687111e+06  268093.728721   
4              SAMUEL PETER    ECLIPSE  2.687111e+06  268093.728721   
...                     ...        ...           ...            ...   
333779     FARBER BETTE LEE     HANSEN  2.715524e+06  276027.686590   
333780        URBINA JUAN P     HANSEN  2.715552e+06  276008.614214   
333781        URBINA JUAN P     HANSEN  2.715552e+06  276008.614214   
333782  1307-09 COTTMAN LLC     HANSEN  2.716305e+06  275494.383618   
333783  1307-09 COTTMAN LLC     HANSEN  2.716305e+06  275494.383618   

       council_di posse_jobi                    geometry  
0               8  195194637  POINT (-75.18310 40.04061)  
1               8  195194637  POINT (-75.18310 40.04061)  
2               8  195194637  POINT (-75.18310 40.04061)  
3               8  195194637  POINT (-75.18310 40.04061)  
4               8  195194637  POINT (-75.18310 40.04061)  
...           ...        ...                         ...  
333779       None       None  POINT (-75.08082 40.06006)  
3337

In [24]:
secondviolations.info()

<class 'geopandas.geodataframe.GeoDataFrame'>
RangeIndex: 333784 entries, 0 to 333783
Data columns (total 31 columns):
 #   Column      Non-Null Count   Dtype   
---  ------      --------------   -----   
 0   objectid    333784 non-null  int64   
 1   addressobj  333784 non-null  object  
 2   parcel_id_  20623 non-null   object  
 3   casenumber  333784 non-null  object  
 4   casecreate  330820 non-null  object  
 5   casecomple  294395 non-null  object  
 6   casetype    20686 non-null   object  
 7   casestatus  333784 non-null  object  
 8   caserespon  330610 non-null  object  
 9   casepriori  330806 non-null  object  
 10  violationn  333784 non-null  object  
 11  violationd  333784 non-null  object  
 12  violationc  333784 non-null  object  
 13  violatio_1  333679 non-null  object  
 14  violations  331473 non-null  object  
 15  violationr  8909 non-null    object  
 16  violatio_2  8909 non-null    object  
 17  mostrecent  324863 non-null  object  
 18  opa_accoun  3313

In [25]:
complaints = gpd.GeoDataFrame.from_file("https://phl.carto.com/api/v2/sql?q=SELECT+*+FROM+complaints&filename=complaints&format=shp&skipfields=cartodb_id")

In [26]:
complaints.info()

<class 'geopandas.geodataframe.GeoDataFrame'>
RangeIndex: 811125 entries, 0 to 811124
Data columns (total 27 columns):
 #   Column      Non-Null Count   Dtype   
---  ------      --------------   -----   
 0   objectid    811125 non-null  int64   
 1   addressobj  810749 non-null  object  
 2   address     810616 non-null  object  
 3   unit_type   727 non-null     object  
 4   unit_num    5798 non-null    object  
 5   zip         810612 non-null  object  
 6   censustrac  810464 non-null  object  
 7   parcel_id_  805063 non-null  object  
 8   opa_accoun  802961 non-null  object  
 9   opa_owner   806119 non-null  object  
 10  complaintn  811125 non-null  object  
 11  casenumber  100788 non-null  object  
 12  complaintc  811125 non-null  object  
 13  complain_1  811125 non-null  object  
 14  unitrespon  164725 non-null  object  
 15  ticket_num  125207 non-null  object  
 16  complaintd  811125 non-null  object  
 17  complaints  164725 non-null  object  
 18  casestatus  1007

In [33]:
# why won't this work
# deprecycling = gpd.read_file("https://www.pasda.psu.edu/json/LandRecyclingCleanupLocations2022_10.geojson")

TypeError: ignored

In [36]:
recyclingdep = gpd.GeoDataFrame.from_file("https://www.pasda.psu.edu/download/dep/LandRecyclingCleanupLocations2022_10.zip")

TypeError: ignored

In [39]:
deptrashland = gpd.GeoDataFrame.from_file("https://www.pasda.psu.edu/json/LandRecyclingCleanupLocations2022_10.geojson")

TypeError: ignored

In [40]:
water = gpd.GeoDataFrame.from_file("https://www.pasda.psu.edu/download/dep/LandRecyclingCleanupLocations2022_10.zip")

TypeError: ignored